### ISSUE TITLE
Autocomplete error Emacs 28

### ISSUE THREAD

**Issue by kstatz12** [fsharp/emacs-fsharp-mode#285]
### Description

When typing f# and triggering autocomplete i get the following error
```
LSP :: Guessed project root is ~/src/csharp-language-server
LSP :: Connected to [fsac:2336053/starting].
LSP :: fsac:2336053 initialized successfully in folders: (/home/karl.statz/src/csharp-language-server)
LSP :: Workspace Loaded!
Quit
fsharp--compute-indentation-relative-to-previous: End position is smaller than start position [2 times]
LSP :: Error from the Language Server: Position is out of range (Internal Error) [4 times]
Company: An error occurred in auto-begin
Company: backend (:separate company-capf company-yasnippet) error "No Typecheck results" with args (candidates Ms)
Company: An error occurred in auto-begin
Company: backend (:separate company-capf company-yasnippet) error "No Typecheck results" with args (candidates MSB)
Company: An error occurred in auto-begin
Company: backend (:separate company-capf company-yasnippet) error "No Typecheck results" with args (candidates MSBu)
```

### Repro steps

Please provide the steps required to reproduce the problem

1. Open F# Project

2. Start writing code

### Expected behavior

Autocomplete suggestions appear

### Actual behavior

No Autocomplete suggestions appear

* Operating system
- Ubuntu 20.04
* Branch
* Emacs version
- 28
* .NET Runtime, CoreCLR or Mono Version
- .net 6

**Comment by amcguier** [comment]
I bumped into a similar issue to this under Emacs 27 (on OSX) which might be helpful. I ended up uninstalling all my pre .NET 6  .NET runtimes / SDKs as there was some interaction between them that caused FsAutoComplete to not finish typechecking.

**Comment by lazywithclass** [comment]
Did you solve it? I have a similar problem but I don't have any previous runtimes around.

**Comment by gorosz** [comment]
I had to reinstall fsautocomplete and it made it work, after lsp workspace restart

`dotnet tool update -g fsautocomplete`

Emacs version: 29.0.50
OS: Fedora release 35 (Thirty Five)
Dotnet version: 6.0.105
Doom emacs version: 3.0.0-dev
Doom modules version: 22.05.0-dev

**Comment by juergenhoetzel** [comment]
> I had to reinstall fsautocomplete and it made it work, after lsp workspace restart
> 
> `dotnet tool update -g fsautocomplete`
> 
> Emacs version: 29.0.50 OS: Fedora release 35 (Thirty Five) Dotnet version: 6.0.105 Doom emacs version: 3.0.0-dev Doom modules version: 22.05.0-dev

I wonder how that fixed the issue:

`eglot-fsharp` uses its own tool path so that there are no conflicts with the tools installed by the user:

```
→ dotnet tool list --tool-path ~/.emacs.d/FsAutoComplete/netcore/
Paket-ID            Version      Befehle       
-----------------------------------------------
fsautocomplete      0.54.0       fsautocomplete
```

**Comment by juergenhoetzel** [comment]
Current `master` branch now works fine with all versions in the build Matrix:
```
        dotnet: [6.0.200]
        emacs_version:
          - 27.1
          - 27.2
          - 28.1
          - snapshot
```

Would you mind to run the integrations tests locally using [eldev](https://github.com/doublep/eldev)
```bash
eldev -d  test
```

**Comment by juergenhoetzel** [comment]
> LSP :: Guessed project root is ~/src/csharp-language-server
> LSP :: Connected to [fsac:2336053/starting].
> LSP :: fsac:2336053 initialized successfully in folders: (/home/karl.statz/src/csharp-language-server)
> LSP :: Workspace Loaded!

It seems you are using `lsp-mode` instead of `eglot-fsharp`  (part of this repo). If this is the case, please open a bug report here:
 [lsp-mode](https://github.com/emacs-lsp/lsp-mode/)

**Comment by juergenhoetzel** [comment]
Should be fixed by #292
